src/pages/EventDetailsPage/index.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+
+import Loading from '../../components/Loading';
+
+import EventDetailsComponent from '../../components/EventDetails';
+import { getEventById } from '../../services/events';
+
+class EventDetailsPage extends React.Component {
+  constructor(props) {
+    const {
+      match: {
+        params: { id },
+      },
+    } = props;
+
+    super(props);
+
+    this.state = {
+      eventId: id,
+      eventInfo: null,
+    };
+  }
+
+  componentDidMount() {
+    const { eventId } = this.state;
+
+    getEventById(eventId)
+      .then(eventObj => {
+        this.setState({ eventInfo: eventObj });
+      })
+      .catch(err => {
+        throw Error(err);
```

Comment:
Does throw Error crash the app? If so just console.log for now - in ideal world we would display error msg to user.